nntrainer/layers/lstm.cpp

```diff
@@ -81,27 +88,31 @@ int LSTMLayer::initialize(Manager &manager) {
                                       WeightRegularizer::NONE, 1.0f, true);
   }
 
-  mem_cell =
-    std::make_shared<Var_Grad>(output_dim[0], true, true, "LSTM:mem_cell");
-  mem_cell->getVariableRef().setZero();
-  mem_cell->getGradientRef().setZero();
-
   TensorDim d = input_dim[0];
-  d.width(unit * NUM_GATE);
+  d.width(unit);
 
+  mem_cell = std::make_shared<Var_Grad>(d, true, true, "LSTM:mem_cell");
+  hidden = std::make_shared<Var_Grad>(d, true, true, "LSTM:temp_hidden");
```

Comment:
in my understanding, hidden stores the output of the each time slice. If yes, how about rename to `LSTM:output`?

Reply:
done